core/static/core/js/emlo-io.js

```diff
@@ -34,4 +34,64 @@ function show_delete_dialog() {
     var retVal = confirm("Delete record?");
     console.log(retVal)
 
-}
\ No newline at end of file
+}
+
+function setup_standtext() {
+    $('.res_standtext').each((i, e) => {
+        append_standtext_ele(e)
+    });
+}
+
+function create_standtext_btn(text, target_id) {
+    let ele = $('<div>');
+    ele.text(text)
+    ele.attr('target_id', target_id)
+    ele.addClass('standtext_btn')
+
+    ele.on('click', (e) => {
+
+        let target_ele = $('#' + ele.attr('target_id'));
+        let space = target_ele.val() ? ' ' : '';
+        target_ele.val(
+            `${target_ele.val()}${space}${ele.text()}`
+        );
+    });
+
+    return ele;
+
+}
+
+function append_standtext_ele(target_text_ele) {
+
+    // build list of standtext button
+    const standtext_list = [
+        'GeoNames', 'TGN', 'Wikidata ID', 'Wikipeda',
+    ];
+    const text_ele_list = standtext_list.map((t) => {
+        return create_standtext_btn(t, target_text_ele.id)
+    })
+
+    //  build standtext container
+    let standtext_div_jqe = $('<div>')
+    standtext_div_jqe.addClass('flex')
+    standtext_div_jqe.append(text_ele_list)
+    $(target_text_ele).after(standtext_div_jqe)
+
+}
+
+
+function setup_checkbox_position() {
+    /** django always put checkbox after the label
+     *  this function reorder checkbox and label, put checkbox before label
+     *  for scss "+ label" style
+     */
+    $('.elcheckbox').each((i, e) => {
+        let checkbox_jqe = $(e)
+        let label = checkbox_jqe.prev('label')
+        checkbox_jqe.parent().remove(checkbox_jqe)
+        label.before(
+            checkbox_jqe
+        )
+    })
+
+}
```

Comment:
Did you consider serverside solution? https://stackoverflow.com/questions/572263/how-do-i-create-a-django-form-that-displays-a-checkbox-label-to-the-right-of-the

Reply:
I did consider add code like this
```
    {% if field.field.widget|is_checkbox %}
      {{ field }}{{ field.label_tag }}
    {% else %}
      {{ field.label_tag }}{{ field }}
    {% endif %}
```
but it will become change form renderer method every time  if form have checkbox 


the javascript could be not the best solution, that is okay to find other solution.